### ISSUE TITLE
How to test for criterion 1.3.6 identify purpose and what would fail it

### ISSUE THREAD

**Issue by DhruvPatel2502** [w3c/wcag#3221]

**Comment by kimviens** [comment]
Thank you for your question Dhruv. I too I am stuck on this, has anyone ever failed a website for this criterion?

**Comment by kimviens** [comment]
I am willing to suggest that this criteria can be testing by using an assistive technology like a screen reader. Screen readers have the ability to pull up a dialog listing all: links, headings, buttons, controls, landmarks... From there we can see if every element can be understood without the context of the content.

**Comment by detlevhfischer** [comment]
Looking at the [Quickref](https://www.w3.org/WAI/WCAG22/quickref/#identify-purpose), the only existing Technique currently provided for 1.3.6 is  [ARIA11](,https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA11.html). This does not seem to make a lot of sense. Landmarks provide structure to non-visual users and do nothing for sighted users, while one main focus of 1.3.5 and 1.3.6 seems to be sighted users who struggle with textual labels. Also, landmarks typically appy to entire regions of content, not to individual inputs (perhaps with the exception of `role="search"` as a container role for elements around search fields).

I guess the most commion approach to satisfy 1.3.6 would be the same as for 1.3.5: using the `autocomplete` attribute as a language-independent way of communicating what kind input is expected. Strangely, there seems to be no such Technique referenced in the Quickref for 1.3.5. Maybe it hasn't been written yet.

The main point of 1.3.5 and 1.3.6 was supposed to provide a generic identification of input purpose independent of, and going beyond, the HTML `type` attribute). The idea was that new assistive technologies (e.g. browser plugins) could use that info to present the input purpose in some other (possibly not texual) form. 

So, @kimviens ,  using the screen reader to test this does not seem correct. There are [bookmarklets checking for the existence of `autocomplete` attributes ](https://a11y-tools.com/bookmarklets/) (no endorsement implied, haven't tried it yet) so the autocomplete values used could be compared to the inputs' respective purpose. 

The basic problem in this "hen and egg" scenario remains that three years after introcuction of the 1.3.5 SC in WCAG 2.1, no AT whatsoever has appeared that makes use of a specific input purpose attribute (currently `autocomplete`) as a stand-in for future more generic metadata attributes. (Correct me if I am wrong and something **does** exist, please!). Having `autocomplete` assist users in filling in fields is often be a welcome side effect but was to the actual purpose of 1.3.5. and 1.3.6.

**Comment by patrickhlauke** [comment]
agree that 1.3.6 is eminently vague. I see it a bit as a stricter version of 1.3.1 Info and Relationships ... while 1.3.1 is scoped just to making sure the underlying structure of a page reflects its visual presentation, 1.3.6 goes further in demanding that the markup should convey often implicit structures in a more unambiguous (if possible) way.

i'd add a bit more nuance here though to @detlevhfischer's initial thoughts. I do think ARIA landmarks do play a role here...while yes, the idea of the SC is that it can benefit sighted users, and ARIA landmarks primarily serve non-sighted/AT users, the SC itself asks for adding "programmatically determinable" information. landmarks satisfy this. The idea then is that a sighted user may be able to have a bookmarklet, custom CSS, some mythical/hypothetical feature in their user agent, that can also make this programmatically determinable structural information visible to them.

but how to determine what passes or fails this SC is...wooly, at best. comes down, in my view, to "is the page using all possible techniques, such as landmarks etc, to give as much information as possible about its structure and the purpose of its various component elements/interactive elements?" ... it's very open-ended

**Comment by detlevhfischer** [comment]
@patrickhlauke 

> I'd add a bit more nuance here though to @detlevhfischer's initial thoughts. I do think ARIA landmarks do play a role here

Yeah - I looked up 1.3.6 and realised the focus of my answer was too narrow, have updated it while you were writing...

**Comment by alastc** [comment]
@detlevhfischer - there are some pluggins which apply icons to the content. I know they exist but are relatively specialist (even within accessibility) and I haven't tried one myself. I don't know if they utilize autocomplete or landmarks, but they should.

I'd also note that landmarks aren't required by 1.3.1 (there are other techniques), but my read of 1.3.6 is that landmarks are generally required (assuming there are different areas of a page).

**Comment by patrickhlauke** [comment]
> I'd also note that landmarks aren't required by 1.3.1 (there are other techniques), but my read of 1.3.6 is that landmarks are generally required (assuming there are different areas of a page).

yup, hence my high-level "1.3.6 is a bit like a stricter version of 1.3.1" gut-feel

**Comment by detlevhfischer** [comment]
@alastc 
> there are some pluggins which apply icons to the content

Ah, OK, I see. Do you have a link to one or any of them?

**Comment by patrickhlauke** [comment]
I think the only plugins/bookmarklets I've seen were those created at the time to justify making the SC pass / demonstrate some "implementation"...

**Comment by patrickhlauke** [comment]
coming back to this after a few months... are we (AGWG) agreed/resigned to the fact that this SC, in practice, currently really just shakes out to "use landmark elements (`<header>`,`<nav>`,`<main>`, etc) or ARIA landmarks (`role="main"`, etc)" ?
